app/models/extensions/order_subscriptions_extensions.rb

```diff
@@ -0,0 +1,14 @@
+# Used to prevent payments on subscriptions from being processed in the normal way.
+# Payments are skipped until after the order cycle has closed.
+module OrderSubscriptionsExtensions
```

Comment:
Dear reviewers,

This file introduces a new structure. I simply didn't know where to put this file. It is not a Concern, but an Extension. It is very tightly coupled to Spree::Order and Subscription logic. So it doesn't fit into `lib`. It's not a service and it's not a model on its own. Is this the right place?

Reply:
The name OrderSubscriptionsExtensions implies we can add more to this module. Modules and classes should have single responsibility and as such I'd name the module to what it is doing here and nothing else: OrderSubscriptionsPaymentRequired or OrderSubscriptionsProcessPayments.

re /extensions, I'd do an effort not to create another type of file in the codebase, it adds complexity that we should avoid.
I dont think it would be perfect, but I think it'd be better than this extension: a concern with a alias_method_chain on  payment_required? or better, process_payments.

I am sure @sauloperez will be thinking about spree hooks. The right solution for this would be to patch spree code to add a "before_process_payments" hook and then make this change in ofn code reimplementing the hook.

Personally, I'd go for the alias_method_chain on process_payments inside a concern.

Reply:
> I am sure @sauloperez will be thinking about spree hooks

I'm too foreseeable :trollface: . That is exactly what I though right after reading the description. I tend to dislike `alias_method_chain` for what I shared in https://github.com/openfoodfoundation/openfoodnetwork/pull/3009#discussion_r233396535.

>  The right solution for this would be to patch spree code to add a "before_process_payments" hook and then make this change in ofn code reimplementing the hook.

Then, why not this? or any other extension hook we think best fits here?

Reply:
With "The right solution" I meant "the solution that @sauloperez thinks is right" :-)
I think we have to explore option 4 below. Try to remove subscriptions code from order.